frontend/src/scenes/error-tracking/queries.ts

```diff
@@ -185,11 +185,8 @@ export const errorTrackingGroupEventsQuery = ({
 
 // JSON.stringify wraps strings in double quotes and HogQL only supports single quote strings
 const stringifyFingerprints = (fingerprints: ErrorTrackingGroup['fingerprint'][]): string => {
-    const stringifiedFingerprints = fingerprints.map((fp) => {
-        const stringifiedParts = fp.map((s) => `'${s}'`)
-        return `[${stringifiedParts.join(',')}]`
-    })
-    return `[${stringifiedFingerprints.join(',')}]`
+    // so we escape all single quoted strings and replace double quotes with single quotes
+    return JSON.stringify(fingerprints).replace(/'/g, "\\'").replace(/"/g, "'")
```

Comment:
@daibhin this was causing issues because fingerprints can have single quotes 🙈 . I think what I have works well, but lmk if you see something wonky here!

Reply:
This feels less brittle 👍 Should the `replace(/"/g, "'")` come before `replace(/'/g, "\\'")` or would that mess up the actual matching? Either way maybe it's worth adding a test on the backend to confirm that escaped strings match as we expect them to

Reply:
if its before, then the outer double quotes would be turned into single quotes, which we don't want, and the inner escaped double quotes that json.stringify creates wouldn't be properly parsed either 😅 .

I thought I added the test, but looks like I only added it for error search, this will go to event query, will add 👍

Reply:
fix was a bit more involved, I didn't realise JSON adds a backslash before double quotes, so we had to remove these in the end as well. Should be ready now though